src/decoder.rs

```diff
@@ -785,12 +785,14 @@ fn compute_image(components: &[Component],
         let height = component.size.height as usize;
 
         let mut buffer = vec![0u8; width * height];
-        let line_stride = width * component.dct_scale;
-
-        for y in 0 .. width {
-            for x in 0 .. height {
-                buffer[y * width + x] = data[0][y * line_stride + x];
-            }
+        let line_stride = component.block_size.width as usize * component.dct_scale;
+
+        for y in 0 .. height {
+            let destination_idx = y * width;
+            let source_idx = y * line_stride;
+            let destination = &mut buffer[destination_idx..destination_idx + width];
```

Comment:
One trick you can do here to avoid repeating destination_idx is to use `buffer[destination_idx..][..width]`